src/session.rs

```diff
@@ -64,6 +68,12 @@ pub struct Session<FS: Filesystem> {
     pub(crate) destroyed: bool,
 }
 
+impl<FS: Filesystem> AsFd for Session<FS> {
+    fn as_fd(&self) -> BorrowedFd<'_> {
+        self.ch.as_fd()
+    }
+}
+
```

Comment:
I think it would be better to have a `mount_to_fd()` function similar to `mount()` that instead returns the FD. That better separates the concerns of creating the mountpoint, which won't even need to have a `Filesystem` implementation.

Reply:
This is just for symmetry - I'm not using it, so I can remove it if you don't like it. Note that the stdlib equivalents all support this (File, various sockets, etc).

Reply:
How would someone use it though?

The reason it doesn't seem right to me is that a `Session` represents a mounted filesystem, borrowing its fd, and reading/writing it would interfere with the `Filesystem` that is mounted and feels like it breaks the encapsulation.

Reply:
> How would someone use it though?

By passing it into `mount(2)`/`fsconfig(2)` to mount the session somewhere.

In my use case, you need to mount first and then create the session. But it's not hard to imagine someone needing to do it the other way around.

Reply:
Zooming out a bit:

I would like to submit that FUSE is a very low-level concept, and it's often used as a hack-on-demand. So people using it probably know what they're doing, and may have complicated requirements. A fuse session is a `/dev/fd` fuse - therefore it fulfills the requirements of `AsFd`.

I do see the concerns around breaking the encapsulation, but `Channel` is not actually stateful, right? So it's really very similar to `UnixStream` or whatever. If you call `AsFd` on a `UnixStream` and drop down to `nix::io::write` or whatever, you probably have a very specific use case that the stdlib authors didn't think of.

Reply:
Good points. Thanks for explaining the use case!